### PR TITLE
Add highlight components

### PR DESCRIPTION
**Pull request by viktoria-mahmud**

### CODE REVIEW

**Review by discodavey — COMMENTED** [review]
@nlisgo Should this be named Highlights as there is more than one showing in the pattern library example?

File: source/_patterns/01-molecules/components/highlight-item.mustache
Comment (discodavey) [inline source/_patterns/01-molecules/components/highlight-item.mustache:24]: I think should be in a paragraph tag.

```<p class="paragraph">```

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:60]: Can we add the padding on the bottom of the class author-line. 

We try and put the margins and padding on the bottom of the classes or id's unless it's impossible.

**Review by discodavey — COMMENTED** [review]
remove id as empty
Screenshot 2022-09-07 at 10 53 44

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:35]: Can we not use % on width please

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:38]: Same as above. Do not use %

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:81]: Can we align this using grid rather than using float?

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:105]: Is this being used anywhere. If not then please remove.

**Review by discodavey — COMMENTED** [review]
@chris-huggins Just a thought... Should we make the image as a link as well?

**Comment by chris-huggins** [comment]
> @chris-huggins Just a thought... Should we make the image as a link as well?

Yes please

**Comment by viktoria-mahmud** [comment]
> @nlisgo Should this be named Highlights as there is more than one showing in the pattern library example?

Curently we have few files named "Highlights" and they are not related with the current component:
image

Do I need to rename some of these files?

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (discodavey) [inline assets/sass/patterns/molecules/highlight-item.scss:136]: Move the padding top from the title and put on the bottom of the image. Should be 24px.

File: source/_patterns/01-molecules/components/highlight-item.mustache
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/components/highlight-item.mustache]: The tag is changed to be a paragraph.

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss]: The padding is added to the author class as padding bottom.

I will add future paddings and margins on the bottom of the classes.

I think in should be 12px padding between the authors line and meta, correct me if I am wrong.

**Comment by viktoria-mahmud** [comment]
> remove id as empty Screenshot 2022-09-07 at 10 53 44

The Id is not used, so I removed it.

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss]: The width with % is removed and the logic is changed to be with grid style.

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss]: The width with % is removed and the logic is changed to be with grid style.

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss]: The float is removed and now the style is aligned as grid

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss]: Thanks, I missed that style, it is removed now

> > @chris-huggins Just a thought... Should we make the image as a link as well?
> 
> Yes please

The images is wraped in a link tag

File: assets/sass/patterns/molecules/highlight-item.scss
Comment (viktoria-mahmud) [reply assets/sass/patterns/molecules/highlight-item.scss:108]: The padding is moved as a padding bottom of the image as 24px.

File: source/_patterns/01-molecules/components/highlight-item.mustache
Comment (nlisgo) [inline source/_patterns/01-molecules/components/highlight-item.mustache:1]: We do not need data-behaviour attribute as we are not assigning any js to this item.

File: source/_patterns/01-molecules/components/highlight-item.yaml
Comment (nlisgo) [inline source/_patterns/01-molecules/components/highlight-item.yaml:11]: `minLength: 1` still applies. If the value is present it should not be empty.

File: source/_patterns/02-organisms/components/highlight.mustache
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight.mustache:1]: If `headingId` is needed then we should add it to the schema. @discodavey?

File: source/_patterns/02-organisms/components/highlight.json
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight.json:1]: We should have variants with 1 and 2 items also. To achieve this we should not use `highlight.json` and instead just encourage the use of variants `highlight~full.json`, `highlight~one-item.json` and `highlight~two-items.json`. We do not need to support more than 3.

File: source/_patterns/01-molecules/components/highlight-item.json
Comment (nlisgo) [inline source/_patterns/01-molecules/components/highlight-item.json:1]: In the same way as hero banner we should have a variant with each different type of content.

**Review by nlisgo — CHANGES_REQUESTED** [review]

**Comment by discodavey** [comment]
> > @nlisgo Should this be named Highlights as there is more than one showing in the pattern library example?
> 
> @discodavey This is a good point. Might be better to rename and avoid confusion. What would you suggest?

Will have a look at what the other one's do and think of a name for them.

File: source/_patterns/02-organisms/components/highlight.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight.json:1]: @chris-huggins can you provide a design for one and two items for all screens

File: source/_patterns/01-molecules/components/highlight-item.json
Comment (nlisgo) [reply source/_patterns/01-molecules/components/highlight-item.json:1]: > @nlisgo can you advise me what are the different content types for the Highlights? Something like that: -blog-article -collection -digest -event -interview -labs-post -podcast-episode -press-package

The list is the same as for hero banner.

File: source/_patterns/02-organisms/components/highlight.json
Comment (nlisgo) [reply source/_patterns/02-organisms/components/highlight.json:1]: @viktoria-mahmud as less than 3 highlights will just be on view on the preview instance of journal then it's not so important to get the design perfect.

Let's just see how they appear when there is one or two highlights and then we can either just agree that is acceptable or suggest a minor adjustment to make it not look terrible.

File: source/_patterns/01-molecules/components/highlight-item.yaml
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/components/highlight-item.yaml:11]: Done

File: source/_patterns/02-organisms/components/highlight.mustache
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight.mustache]: I removed the headingId, it is not needed

File: source/_patterns/02-organisms/components/highlight~article-long-authors.json
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight~article-long-authors.json:21]: The variants for highlight should be against highlight-item.

**Review by nlisgo — CHANGES_REQUESTED** [review]

File: source/_patterns/02-organisms/components/highlight~one-item.json
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight~one-item.json:7]: You should use an example that looks more real.

File: source/_patterns/02-organisms/components/highlight~two-items.json
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight~two-items.json:21]: You should use actual examples.

File: source/_patterns/02-organisms/components/highlight.mustache
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight.mustache:6]: The three columns style is achieved based on the items count, if I remove that itemsCount, I can't say when we have one or three elements and the alignment is not very pleasent.

File: source/_patterns/02-organisms/components/highlight~article-long-authors.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight~article-long-authors.json:21]: If these variants are against highlight-item, I can add only one item for each type. Is that fine?

File: source/_patterns/02-organisms/components/highlight.mustache
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight.mustache]: The itemsCount is removed

File: source/_patterns/02-organisms/components/highlight~article-long-authors.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight~article-long-authors.json]: Done, all examples are created as a highlight-item

File: source/_patterns/02-organisms/components/highlight~one-item.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight~one-item.json]: Real data is added in all examples for one, two and full items

File: source/_patterns/02-organisms/components/highlight~two-items.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight~two-items.json]: Real data is added in all examples for one, two and full items

**Comment by discodavey** [comment]
Margin top is not correct on title at 900 breakpoint

Screenshot 2022-09-09 at 10 58 48

**Comment by discodavey** [comment]
Another ticket will need to be made for the authors all the authors across the site from hero banner, hightlights, teasers and in the content headers should be consistent and should be either in a div or a list.

For best practice I think the authors should be in a list like it is in the content header on articles. David M agrees with me on this so can be done as another piece of work.

**Comment by viktoria-mahmud** [comment]
> Margin top is not correct on title at 900 breakpoint and below
> 
> Screenshot 2022-09-09 at 10 58 48

@chris-huggins do we need margin above the title? There is some small space above the highlight title but it seems like a mistake. If it's not, I will fix the code

**Comment by nlisgo** [comment]
> Another ticket will need to be made for the authors all the authors across the site from hero banner, hightlights, teasers and in the content headers should be consistent and should be either in a div or a list.
> 
> For best practice I think the authors should be in a list like it is in the content header on articles. David M agrees with me on this so can be done as another piece of work.

@discodavey Are you going to raise this ticket? The authors are consistent across all representations of authors in hero banner, highlights and teasers. When you raise that ticket can you give an example of how the markup should look also?

File: source/_patterns/01-molecules/components/highlight-item~article-long-authors.json
Comment (nlisgo) [inline source/_patterns/01-molecules/components/highlight-item~article-long-authors.json:2]: Please change to an actual title.

File: source/_patterns/01-molecules/components/highlight-item~article-long-authors.json
Comment (nlisgo) [inline source/_patterns/01-molecules/components/highlight-item~article-long-authors.json:4]: Could this be changed to something like:

```
"authors": "Thisisanexample Ofareallylongauthorline, Todemonstrate Whatitlookslike ... Ifthis Valuewraps",
```

File: source/_patterns/02-organisms/components/highlight~one-item.json
Comment (nlisgo) [inline source/_patterns/02-organisms/components/highlight~one-item.json:6]: @viktoria-mahmud I think it looks great with 1 and 2 items. Good enough anyway. We will be insisting that 3 highlights for the homepage are recommended. If ever this changes then we can revisit how they are appearing with fewer items than 3.

**Review by nlisgo — APPROVED** [review]
Approving if the small changes I have requested here are made. Great work @viktoria-mahmud

File: source/_patterns/01-molecules/components/highlight-item~article-long-authors.json
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/components/highlight-item~article-long-authors.json]: Done

File: source/_patterns/01-molecules/components/highlight-item~article-long-authors.json
Comment (viktoria-mahmud) [reply source/_patterns/01-molecules/components/highlight-item~article-long-authors.json]: Done

File: source/_patterns/02-organisms/components/highlight~one-item.json
Comment (viktoria-mahmud) [reply source/_patterns/02-organisms/components/highlight~one-item.json:6]: > Approving if the small changes I have requested here are made. Great work @viktoria-mahmud

The changes are ready. Thank you!

**Comment by nlisgo** [comment]
@viktoria-mahmud there are merge conflicts to resolve on:

- source/_patterns/04-pages/home.json
- source/_patterns/04-pages/home.mustache

Let me know if you need any assistance with that.

@discodavey after merge conflicts are resolved can you confirm that all your feedback has been addressed and then merge?

**Comment by viktoria-mahmud** [comment]
> @viktoria-mahmud there are merge conflicts to resolve on:
> 
> * source/_patterns/04-pages/home.json
> * source/_patterns/04-pages/home.mustache
> 
> Let me know if you need any assistance with that.
> 

@nlisgo The conflicts are resolved

**Comment by discodavey** [comment]
@nlisgo @viktoria-mahmud Can you both just check to make sure it looks ok and the conflicts have been sorted ok?

If they have then I can merge it

**Comment by viktoria-mahmud** [comment]
> @nlisgo @viktoria-mahmud Can you both just check to make sure it looks ok and the conflicts have been sorted ok?
> 
> If they have then I can merge it

Yes, it is fine now

**Pull request merged**
